bluemira/optimisation/_nlopt/functions.py

```diff
@@ -94,7 +94,10 @@ def _call_inner(self, x: np.ndarray, grad: np.ndarray) -> float:
         # Cache f(x) so we do not need to recalculate it if we're using
         # an approximate gradient
         self.f0 = self.f(x)
+        print(grad, flush=True)
         if grad.size > 0:
+            print(x, grad, self.f0, self.df(x), flush=True)
+            grad[:] = 0
```

Comment:
So this does work then?

Reply:
Oh perhaps not... was the failure only on 3.11 originally?

Reply:
yeah its always been a failure on 3.11, It doesn't seem to be the same thing as we talked about the other day...

Reply:
Ah OK, that's a shame I guess..